oeps/oep-0028-content-theming-in-react.rst

```diff
@@ -0,0 +1,279 @@
+-------
+
+Open edX instances frequently wish to customize not only the colors and styles of the Open edX UI, but also to add, remove, and modify significant parts of the UI to suit their use cases and branding. Doing this in a way that is understandable, clean, and easy to maintain can be a challenge unless the original UI is designed with this in mind.
+
+Another pain of theming systems is providing access to unrelated data to any component. It becomes important to make commonly-used data globally available to all components.
+
+Decision
+--------
+
+* We will build two types of components:
+
+  1. Customizable components which will be composed of others components
+
+  2. Internal or non-customizable components which will contain HTML and certain logic
+
+* We will build small modular components which will be non-customizable, stateless, contain HTML, and certain logic. A component will be defined as non customizable if it is only concerned with how things look and does not maintain any state of its own. In case they do, it will be UI state rather than data. They will be generally written as functional components, unless they need any data state or any lifecycle hook. They will receive data and callbacks via props and will have no dependencies on the rest of application. We can see an example of a smaller component ``SiteLogo``, which will later be used in a customizable component ``_Header``.
+
+.. code-block:: js
+
+    const SiteLogo = (props) => {
+        return (
+            <img src={props.url} alt={props.altText} />
+        );
+    }
+
+* We will build the complete UI out of various customizable components. These customizable components will contain small modular components, little or no HTML nor logic. A component will be termed as customizable if it contains various non customizable and reusable components. Customizable components will be concerned with how the things work and will generally maintain state. They will collect data and define callbacks, which will be passed to non customizable components via props. We can see an example of a customizable component called the ``_Header`` component, which is formed by the composition of ``SiteLogo``, ``MainNav`` and ``UserAvatar`` component.
+
+.. code-block:: js
+
+    class _Header extends React.PureComponent {
+        render() {
+            return (
+                <header>
+                    {this.props.logo ? this.props.logo: <SiteLogo/>}
+                    <MainNav/>
+                    <UserAvatar/>
+                </header>
+            );
+        }
+    }
+
+* We will specify what are the customizable components, document whether the customization is available via custom props or overriding superclass method. We will announce breaking changes if any changes are made to the component. We will prefer that customizable components be only customized, however it is totally up to to the theme owner to customize any components that is not marked as customizable by edX.
```

Comment:
I suggest changing this to:

"We will specify which components are customizable and document whether the customization is available via custom props or overriding superclass methods. We will announce breaking changes if any changes are made to customizable components. Users will still be able to override components which are not explicitly marked as customizable, but edX will provide no backwards compatibility guarantees for them."